elfi/methods/bo/utils.py

```diff
@@ -162,3 +162,67 @@ def evaluate_gradient(self, x):
         x = np.atleast_2d(x)
         n, input_dim = x.shape
         return self.scale * self.gradient(x).reshape(n, input_dim)
+
+
+def make_additive_acq(acquisition_class, function):
+    """Make acquisition function adjusted with an additive term.
+
+    Parameters
+    ----------
+    acquisition_class : Type[elfi.methods.bo.acquisition.AcquisitionBase]
+        Acquisition function to be adjusted.
+    function : AdjustmentFunction
+        Function added to the base acquisition function.
+
+    Returns
+    -------
+    AdditiveAcquisition
+
+    """
+    class AdditiveAcquisition(acquisition_class):
```

Comment:
`AdditiveAcquisition` is inside `make_additive_acq` because the idea is that `make_additive_acq` makes a new acquisition class that inherits the base acquisition class provided as input.

if your comment was about the new acquisition class name, i can rename both `AdditiveAcquisition` and `MultiplicativeAcquisition` to `AdjustedAcquisition`. the current names are indeed not clear and can be misunderstood.

if the comment was about the approach in general, the reason i used dynamic inheritance is that our available acquisition classes have varied implementations for the `acquire` method and the adjusted acquisition class needs to use the same `acquire` method as the base class. but maybe there is a better way to do this?

Reply:
Fair enough, it's an implementation that works for this case. We just have to be careful where and when  it's being called. I guess it generates a new class every time it's being called? Even with the same `AcquisitionBase`-instances?

Reply:
yes, new class every time. i understand that creates some overhead compared to predetermined classes, but i assumed that's fine since we don't call this method more than once per inference task.

Reply:
Yes! Just have  to make sure that users won't try out silly things with it :)